### DISCUSSION TITLE
New feature for http server: get a snapshot

### DISCUSSION THREAD

**Discussion by appliko** [Sunoo/homebridge-camera-ffmpeg#1004] [Ideas]
I'm using a camera where I can connect via rtsp but I cannot get frames via http nor ftp
I need to get frames via http to view images on my Fritz doorcam interface
It would be nice and hopefully not too complex if plugin http server would have a function to get a frame 
Something like http://hostname:port/snap?CameraName

**Comment by ciriguaya** [comment]
I would need something like this as well for another Raspberry Pi to access the video stream via a http address. Would very much appreciate such a plug-in!

**Comment by Sunoo** [comment]
I’m thinking about this, it’s definitely possible to implement, but I’m not sure it makes sense to?

Could someone tell me their use case for this, as opposed to just calling FFmpeg to get a frame themselves?

**Comment by appliko** [comment]
Nello homebridge plugin requires still image to show on call
Also home bridge is showing still frames
fritz box can show http or images on door com function, not rtsp

it is way faster if still frames are available.
As a workaround I have a task on the pi that is calling ffmpeg to save frames on a jpg on a ram disk shared via http …. 
It would be lovely if your plugin just do it

thanks

**Reply by Sunoo** [reply]
I need to do a rewrite of the way I’m currently handling HTTP and MQTT soon. At the point I do that, I will take a look to see if this can be done easily, and may add this at that time.

No promises, but I’m not ruling it out. It would only ever be stills though, I will not be adding any features to handle video over HTTP.
